### ISSUE TITLE
Add support for retrieving data from Unsupported fields in Prisma query using cast to string

### ISSUE THREAD

**Issue by MurzNN** [prisma/prisma#18544]
## Problem

We can describe `Unsupported` field types in Prisma schema, eg `Point` field type like this:
```js
model Locality {
  id         String                @id @default(cuid())
  name       String?
  point      Unsupported("Point")?
}
```
And we can write data to them, using `$executeRawUnsafe`.

But to read the data we should also always use `$queryRaw` which is very inconvenient, because read queries are often very complex and retrieve data from many fields, including relations.

## Suggested solution

We can use SQL Expressions to cast most of the `Unsupported` data types to a simple string on SQL server side, like this:
```sql
SELECT id, ST_AsText(point) point
FROM public."Point";
```
So will be good to add the ability to do this via Prisma queries.

**Comment by guyromb** [comment]
Any update on this?

**Comment by hzgand** [comment]
this is an excellent idea

**Comment by daweimau** [comment]
This would take away SO much work.  Suddenly working with postgis unsupported types would become totally manageable. Currently it is possible, but quickly becomes more and more tedious, especially with relations

**Comment by shadoath** [comment]
Yes, with this, we would be able to keep our other defined fields and types and not need to build the SQL by hand to be able to work with [daterange](https://github.com/prisma/prisma/issues/3287) and other unsupported fields. 
Is there any direction for how this would be implemented? Linking us to the right place in the engine to implement the changes would help shape this.

**Comment by suhjohn** [comment]
what's the status on this?

**Comment by PerishableDave** [comment]
Bump. This would be a substantial quality of life improvement for Prisma.

**Comment by webmobiles** [comment]
Prisma with support for simple embeddins pgvector is important

**Comment by kaikun213** [comment]
This would be great! Eg for ltree or any other unsupported types in pg
